classes/Product.php

```diff
@@ -7176,7 +7176,9 @@ public function setWsPositionInCategory($position)
         // result is indexed by recordset order and not position. positions start at index 1 so we need an empty element
         array_unshift($result, null);
         foreach ($result as &$value) {
-            $value = $value['id_product'];
```

Comment:
I don't understand why `$value` can be null sometimes ? Isn't that the bug that we should fix ? (while your php check is still a good addition)

Reply:
`array_unshift($result, null);` Here is why the value is null the line just above.

Preprending the array with a null value and right after trying to access a key on a null object.

Maybe if it's really needed we should move that array_unshift after that foreach ?